modules/host/contractmanager/persist.go

```diff
@@ -162,5 +191,9 @@ func (cm *ContractManager) savedSettings() savedSettings {
 			sf.setUsage(sectorIndex)
 		}
 	}
+
+	sort.Slice(ss.StorageFolders, func(i, j int) bool {
+		return ss.StorageFolders[i].Index < ss.StorageFolders[j].Index
+	})
```

Comment:
should have a comment like:

```go
// canonicalize storage folder ordering; otherwise savedSettings.equals
// could return false for equivalent settings
```

Reply:
I'll also note that this is the most "dangerous" part of the diff, because it affects the data being written to disk. I'm confident that this won't cause any breakage, but if we were being *really* conservative, we'd probably want to perform the sort in `equals` (after copying the memory, to prevent side effects).